src/Formik.tsx

```diff
@@ -598,8 +599,11 @@ export function useFormik<Values = object>({
               dispatch({ type: 'SUBMIT_SUCCESS' });
             }
           })
-          .catch(_errors => {
+          .catch(error => {
             if (isMounted.current) {
+              if (error instanceof FormikSubmitError) {
+                dispatch({ type: 'SET_ERRORS', payload: error.errors });
```

Comment:
So this will get overwritten on the next key stroke. We should either create a new state key just for this error or use `status.submitError`

Reply:
My idea is to provide a simple way to handle server side validation errors. Since submit handler is a `Promise` now, we could leverage rejected promise state.  I'm all into improving this.

Can you elaborate what do you mean by using `status.submitError`? 

About the first proposal I don't think having another key makes much sense as we already have `SET_ERRORS` and `SUBMIT_FAILURE` and having ie. `SUBMIT_FAILURE_ERRORS` would be redundant. My other idea was to add optional payload (with errors) to `SUBMIT_FAILURE`, bu I'd rather separate setting form errors and submission state. What do you think?

Reply:
My point is just that `errors` gets overwritten on the very next key stroke because validation will run. If you, the dev, want to persist these API errors in your UI until the next submission, then `errors` can't be used to store this data for the aforementioned reason.